src/voterData/voterFile/util/voterFile.util.ts

```diff
@@ -0,0 +1,341 @@
+    "Residence_Addresses_Zip"`
+  }
+
+  if (type === 'sms') {
+    columns += `, "VoterTelephones_CellPhoneFormatted"`
+    if (whereClause) {
+      whereClause += ` AND "VoterTelephones_CellPhoneFormatted" IS NOT NULL`
+    } else {
+      whereClause += `"VoterTelephones_CellPhoneFormatted" IS NOT NULL`
+    }
+  }
+  if (type === 'digitalAds') {
+    columns += `, "VoterTelephones_CellPhoneFormatted",
+    "Residence_Addresses_AddressLine", 
+    "Residence_Addresses_ExtraAddressLine", 
+    "Residence_Addresses_HouseNumber",
+    "Residence_Addresses_City", 
+    "Residence_Addresses_State", 
+    "Residence_Addresses_Zip"`
+
+    whereClause += ` AND "VoterTelephones_CellPhoneFormatted" IS NOT NULL`
+  }
+
+  if (type === 'directMail') {
+    columns += `, "Mailing_Addresses_AddressLine", 
+    "Mailing_Addresses_ExtraAddressLine", 
+    "Mailing_Addresses_City", 
+    "Mailing_Addresses_State", 
+    "Mailing_Addresses_Zip", 
+    "Mailing_Addresses_ZipPlus4", 
+    "Mailing_Families_HHCount"`
+
+    // unique households
+    // whereClause += ` AND "Mailing_Families_FamilyID" IN (
+    //   SELECT "Mailing_Families_FamilyID"
+    //   FROM public."Voter${state}"
+    //   GROUP BY "Mailing_Families_FamilyID"
+    //   HAVING COUNT(*) = 1
+    // )`;
+
+    // measure performance after index is added
+
+    nestedWhereClause = 'a'
+    if (whereClause !== '') {
+      whereClause += ' AND '
+    }
+
+    whereClause += ` EXISTS (
+      SELECT 1
+      FROM public."Voter${state}" b
+      WHERE a."Mailing_Families_FamilyID" = b."Mailing_Families_FamilyID"
+      GROUP BY b."Mailing_Families_FamilyID"
+      HAVING COUNT(*) = 1
+    )`
+  }
+
+  if (type === 'telemarketing') {
+    columns += `, "VoterTelephones_LandlineFormatted",
+    "Languages_Description"`
+
+    whereClause += ` AND "VoterTelephones_LandlineFormatted" IS NOT NULL`
+  }
+
+  if (customFilters?.filters && customFilters.filters.length > 0) {
+    /*
+     custom filter format:
+     {
+      channel: "Door Knocking"
+      filters: ['audience_superVoters', 'audience_likelyVoters', 'party_independent', 'age_18-25', 'age_25-35']
+      purpose: "GOTV"
+  }
+    */
+    whereClause += customFiltersToQuery(customFilters.filters)
+  }
+
+  if (justCount) {
+    return `SELECT COUNT(*) FROM public."Voter${state}" ${nestedWhereClause} ${
+      whereClause !== '' ? `WHERE ${whereClause}` : ''
+    }`
+  }
+
+  return `SELECT ${columns} FROM public."Voter${state}" ${nestedWhereClause} ${
+    whereClause !== '' ? `WHERE ${whereClause}` : ''
+  }`
+}
+
+function extractLocation(input: string, fixColumns?: boolean) {
+  logger.log(
+    `Extracting location from: ${input} ${
+      fixColumns ? '- with fixColumns' : ''
+    }`,
+  )
+  // Remove any trailing '##' from the input string
+  const extracted = input.replace(/##$/, '')
+
+  // Split the string by '##', take the last element, and remove ' (EST.)' if present
+  // if fixColumns is true, we want to use the second element
```

Comment:
Again, this is just writing in a comment what the logic already says it's doing, no? 🤔